### DISCUSSION TITLE
Do I need an ARM binary distribution when using the build.ps1 script

### DISCUSSION THREAD

**Discussion by jhangeerali** [cefsharp/CefSharp#3549] [Q&A]
I built and compiled a version of CEF/Chromium using the flags --minimal-distrib --client-distrib and now I am having trouble building the cef-binary packages because I do not have the ARM binary distribution.

I am building CEF version cef-binary-89.0.17-ge7bbb1d-chromium-89.0.4389.114 with proprietary codecs enabled.

**Comment by amaitland** [comment]
The 89 branch doesn't support ARM64 so you can just edit build.ps1 and comment out the ARM64 lines.

**Reply by jhangeerali** [reply]
gotcha, thanks for the help

**Reply by mediaexplorer74** [reply]
> gotcha, thanks for the help

Hi!

Di you have the compiled arm (arm32, not 64) binaries of CEF? I need them very much to create some EDGE browser's minimal analogue for my old sweet Lumia 950 (os - Windows 10 Mobile, ARM)

**Reply by amaitland** [reply]
CEF doesn't support ARM32 on Windows. WinForms and WPF from memory only work natively on ARM64. CefSharp is built with Visual C++ which only supports ARM64. 

Basically ARM32 is not supported as none of the dependencies support ARM32.

**Reply by mediaexplorer74** [reply]
> CEF doesn't support ARM32 on Windows. WinForms and WPF from memory only work natively on ARM64. CefSharp is built with Visual C++ which only supports ARM64.
> 
> Basically ARM32 is not supported as none of the dependencies support ARM32.

Hi. Thank you for yours answer and your great work.  However, I can normally compile UWP ARM32 projects via VC++ (via Visual Studio 20xx Community) usings it great cross-compiler tool. 
Simply I try to find some man in planet Earth who imported CMake job file into Visual Studio, then fix the compile flags, and re-compiled CEF sources for ARM platform... =)

**Reply by amaitland** [reply]
> However, I can normally compile UWP ARM32 projects via VC++ (via Visual Studio 20xx Community) usings it great cross-compiler tool.

CEF cannot be run under UWP as the required APIs aren't supported. See https://magpcss.org/ceforum/viewtopic.php?f=6&t=16135

CefSharp is built using mixed mode CLI/C++, requires the Visual C++ 2019 runtime, there is no redistribution available from Microsoft for ARM32. 

https://docs.microsoft.com/en-US/cpp/windows/latest-supported-vc-redist?view=msvc-170#visual-studio-2015-2017-2019-and-2022

> CMake job file into Visual Studio, then fix the compile flags, and re-compiled CEF sources for ARM platform... =)

Far as I'm aware chromium doesn't support ARM32 so it's unlikely to be that simple. Best of luck to you.

**Reply by mediaexplorer74** [reply]
> > However, I can normally compile UWP ARM32 projects via VC++ (via Visual Studio 20xx Community) usings it great cross-compiler tool.
> 
> CEF cannot be run under UWP as the required APIs aren't supported. See https://magpcss.org/ceforum/viewtopic.php?f=6&t=16135
> 
> CefSharp is built using mixed mode CLI/C++, requires the Visual C++ 2019 runtime, there is no redistribution available from Microsoft for ARM32.
> 
> https://docs.microsoft.com/en-US/cpp/windows/latest-supported-vc-redist?view=msvc-170#visual-studio-2015-2017-2019-and-2022
> 
> > CMake job file into Visual Studio, then fix the compile flags, and re-compiled CEF sources for ARM platform... =)
> 
> Far as I'm aware chromium doesn't support ARM32 so it's unlikely to be that simple. Best of luck to you.

Oh, thank you for great info links/urls, and  for clever description of CEF's ARM32 poring impossibility. So, mission impossible :(
